CMakeLists.txt

```diff
@@ -135,28 +131,28 @@ if(PLUGIN_UPDATER_GPU)
   if(NOT MSVC)
     set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-Xcompiler -fPIC")
   endif()
-  set(CUDA_SOURCES
-    plugin/updater_gpu/src/updater_gpu.cu
-    plugin/updater_gpu/src/gpu_hist_builder.cu
-    )
-  # use below for forcing specific arch
-  cuda_compile(CUDA_OBJS ${CUDA_SOURCES} ${CUDA_NVCC_FLAGS})
-  
 
+  cuda_add_library(gpuxgboost ${CUDA_SOURCES} STATIC)
+  target_link_libraries(gpuxgboost nccl)
+  list(APPEND LINK_LIBRARIES gpuxgboost) 
```

Comment:
I suspect this would cause libxgboost4j to have a dependency on the *shared* library, which is undesirable because in that case we'd need to compile it separately for Windows/Linux/OS X, bundle in the JAR and then unpack&load during runtime.  Could you confirm this is not the case?